src/dotnet/commands/dotnet-install/PackageToProjectFileAdder.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.DotNet.ExecutablePackageObtainer;
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.Cli
+{
+
+    internal class PackageToProjectFileAdder : ICanAddPackageToProjectFile
+    {
+        public void Add(FilePath projectPath, string packageId)
+        {
+            if (projectPath == null) throw new ArgumentNullException(nameof(projectPath));
+            if (packageId == null) throw new ArgumentNullException(nameof(packageId));
+
+            var argsToPassToRestore = new List<string>
+            {
+                projectPath.Value,
+                "package",
+                packageId,
+                "--no-restore"
+            };
+
+            var command = new DotNetCommandFactory(alwaysRunOutOfProc: true)
```

Comment:
HELP NEEDED: It failed when run in proc, It is worst for this case, I should directly use nuget.xplt.dll, however, the way we carry nuget.xplt.dll cause it failed during test. And I am not confident it is fine in production.